app/lib/core/view_models/account_edit_view_model.dart

```diff
@@ -6,13 +8,48 @@ import 'package:app/core/services/service_locator.dart';
 import 'package:app/core/view_models/base_view_model.dart';
 import 'package:app/ui/common/view_state.dart';
 import 'package:flutter/material.dart';
+import 'package:image_picker/image_picker.dart';
+import 'package:permission_handler/permission_handler.dart';
+import 'package:uuid/uuid.dart';
 
 class AccountEditViewModel extends BaseViewModel {
   final DatabaseService _databaseService = locator<DatabaseService>();
   final NavigationService _navigationService = locator<NavigationService>();
   final DialogService _dialogService = locator<DialogService>();
 
+  final _picker = ImagePicker();
+
   Transporter _account;
+  File _selectedImage;
+  File get selectedImage => _selectedImage;
+  String _imageUrl;
+  String get imageUrl => _imageUrl;
+
+  Future<void> selectImage() async {
+    await Permission.photos.request();
+    var permissionStatus = await Permission.photos.status;
+
+    if (permissionStatus.isGranted) {
+      print('granted');
+      var tempImage = await _picker.getImage(source: ImageSource.gallery);
+
+      if (tempImage != null) {
+        _selectedImage = File(tempImage.path);
+
+        notifyListeners();
+      }
+    } else {
+      _dialogService.showDialog(
+        title: 'Access Request',
+        description: 'Permission to access gallery denied',
+      );
+    }
+  }
+
+  Future<void> setImageUrl() async {
+    _imageUrl =
+        await _databaseService.uploadAvatarImage(_selectedImage, Uuid().v4());
```

Comment:
- [ ] We should not use Uuid here. There is a great possibility of two people uploading different images with same name, and we also want to avoid duplicate images if possible

I was working on this, and figure something like the following would work:
1. The filename should be the string encoding of the image (prevent exact dupes of the image):
```
base64Encode(imageFile.readAsBytesSync())
```
2. We should check if the image doesn't already exist, and if it does, just use it's URL:
```
  @override
  Future<String> getAtrImage(String fileName) async => FirebaseStorage.instance
      .ref()
      .child('atrImages/$fileName')
      .getDownloadURL();
```
3. An easier option for you though, is to just name the avatar image after the transporter's ID, because the ID is always unique! We would need to allow file overwrites in the storage rules.

Reply:
I like the sound of number 3, this make sense that display profile image should be 1 only. This works in my use case.